aws/asg.go

```diff
@@ -258,16 +258,39 @@ func describeTargetGroups(elbv2svc elbv2iface.ELBV2API) (map[string]struct{}, er
 }
 
 // map the target group slice into specific types such as instance, ip, etc
-func categorizeTargetTypeInstance(elbv2svc elbv2iface.ELBV2API, allTGARNs []string) (map[string][]string, error) {
-	targetTypes := make(map[string][]string)
+func categorizeTargetTypeInstance(elbv2svc elbv2iface.ELBV2API, allTGARNs []string) (map[string][]TargetGroupWithLabels, error) {
+	targetTypes := make(map[string][]TargetGroupWithLabels)
 	err := elbv2svc.DescribeTargetGroupsPagesWithContext(context.TODO(), &elbv2.DescribeTargetGroupsInput{},
 		func(resp *elbv2.DescribeTargetGroupsOutput, lastPage bool) bool {
 			for _, tg := range resp.TargetGroups {
 				for _, v := range allTGARNs {
 					if v != aws.StringValue(tg.TargetGroupArn) {
 						continue
 					}
-					targetTypes[aws.StringValue(tg.TargetType)] = append(targetTypes[aws.StringValue(tg.TargetType)], aws.StringValue(tg.TargetGroupArn))
+					var podlabel, podnamespace string
+					log.Debugf("Looking for tags on %s", aws.StringValue(tg.TargetGroupArn))
+					out, err := elbv2svc.DescribeTags(&elbv2.DescribeTagsInput{ResourceArns: []*string{tg.TargetGroupArn}})
+					if err != nil {
+						log.Errorf("cannot describe tags on target group: %v", err)
```

Comment:
We always start uppercase for logs "Failed to describe...".

Reply:
Does it make sense to continue here, because otherwise you work with empty string below the else.